Add classpath for all sourceSets to Eclipse by default (#14534)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc

. Run `gradle wrapper --gradle-version {gradleVersion}` to update the project to {gradleVersion}.
. Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.

[[changes_6.8]]
== Upgrading from 6.7

=== Potential breaking changes

> ==== Eclipse plugin has updated classpath

Reviewer: ```suggestion
==== Projects imported into Eclipse now include custom source set classpaths
```

> Previously, the default Eclipse classpath was calcuated from a fixed set of dependency configurations: `compileClasspath`, `runtimeClasspath`,`testCompileClasspath`, and `testRuntimeClasspath`.

Reviewer: ```suggestion
Previously, projects imported by Eclipse only included dependencies for the main and test source sets. The compile and runtime classpaths of custom source sets were ignored.
```

> Since Gradle 6.8, the the default classpath automatically contains all `xxxCompileClasspath` and `xxxRuntimeClasspath` for every custom
> source sets defined by the build.

Reviewer: ```suggestion
Since Gradle 6.8, projects imported into Eclipse include the compile and runtime classpath for every source set defined by the build.
```
